tests/test_registry.py

```diff
@@ -0,0 +1,34 @@
+import os
+from urllib.parse import urlsplit
+
+from aiidalab.fetch import GitRepo, fetch_from_url
+from aiidalab.registry.releases import _get_release_commits, _split_release_line
+
+
+def test_get_all_tagged_releases():
+    """Test that all tagged releases are returned."""
+    url = "git+https://github.com/aiidalab/aiidalab-qe.git@*:v23.04.0^.."
```

Comment:
Thanks for adding the test! :clap:  
It's not great that we rely here on an external URL, but at least we'll find out if we ever deleted that branch by accident. :D